src/hoi4_world/states/hoi4_states_converter.cpp

```diff
@@ -814,43 +790,55 @@ void LogManpowerStats(const std::vector<hoi4::State>& hoi4_states,
+       infrastructure_mapper.GetConvertedInfrastructure(),
+       infrastructure_mapper.GetTargetInfrastructure(),
+       static_cast<float>(infrastructure_mapper.GetConvertedInfrastructure()) /
+           static_cast<float>(infrastructure_mapper.GetTargetInfrastructure()) * 100.0F);
+   Log(LogLevel::Info) << fmt::format("\tfudge factor is {}", infrastructure_mapper.GetFudgeFactor());
 }
 
 
-hoi4::States CreateStates(const std::map<int, vic3::State>& vic3_states,
-    const vic3::Buildings& vic3_buildings,
+hoi4::States CreateStates(const vic3::World& source_world,
+    const mappers::WorldMapper& world_mapper,
+    const hoi4::WorldFramework& world_framework,
     const std::map<int, std::set<int>>& vic3_state_id_to_hoi4_provinces,
     const std::vector<int>& vic3_state_ids_by_vic3_industry,
-    const mappers::ProvinceMapper& province_mapper,
     const maps::MapData& map_data,
     const maps::ProvinceDefinitions& hoi4_province_definitions,
-    const hoi4::StrategicRegions& strategic_regions,
-    const mappers::CountryMapper& country_mapper,
-    const hoi4::StateCategories& state_categories,
-    const std::map<int, hoi4::DefaultState>& default_states,
-    const std::map<std::string, vic3::StateRegion>& vic3_state_regions,
     const hoi4::CoastalProvinces& coastal_provinces,
-    const hoi4::ResourcesMap& resources_map,
     bool debug)
 {
    std::vector<hoi4::State> hoi4_states;
    std::map<int, int> province_to_state_id_map;
    std::map<int, int> vic3_state_ids_to_hoi4_state_ids;
    std::unordered_map<std::string, FactoriesStruct> accumulator;
 
-   mappers::Hoi4ToVic3ProvinceMapping hoi4_to_vic3_province_mappings = province_mapper.GetHoi4ToVic3ProvinceMappings();
-   mappers::Vic3ToHoi4ProvinceMapping vic3_to_hoi4_province_mappings = province_mapper.GetVic3ToHoi4ProvinceMappings();
-   const std::set<int> wasteland_provinces = GetWastelandProvinces(default_states);
-   std::map<std::string, std::string> significant_provinces = GetAllSignificantProvinces(vic3_state_regions);
-   std::map<std::string, std::string> vic3_provinces_to_state_names = MapVic3ProvincesToStateNames(vic3_state_regions);
+   mappers::Hoi4ToVic3ProvinceMapping hoi4_to_vic3_province_mappings =
+       world_mapper.province_mapper.GetHoi4ToVic3ProvinceMappings();
+   mappers::Vic3ToHoi4ProvinceMapping vic3_to_hoi4_province_mappings =
+       world_mapper.province_mapper.GetVic3ToHoi4ProvinceMappings();
+   const std::set<int> wasteland_provinces = GetWastelandProvinces(world_framework.default_states);
+   std::map<std::string, std::string> significant_provinces =
+       GetAllSignificantProvinces(source_world.GetStateRegions());
+   std::map<std::string, std::string> vic3_provinces_to_state_names =
+       MapVic3ProvincesToStateNames(source_world.GetStateRegions());
    std::map<std::string, std::string> hoi4_state_names_to_vic3_state_names;
+   mappers::InfrastructureMapper infrastructure_mapper(source_world.GetStates());
```

Comment:
Ideally, I think this should be in the world_mapper, but it can't because it contains nonconst members